contracts/contracts/poolBooster/PoolBoosterMerkl.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IPoolBooster } from "../interfaces/poolBooster/IPoolBooster.sol";
+import { IMerklDistributor } from "../interfaces/poolBooster/IMerklDistributor.sol";
+
+interface IERC1271 {
+    /**
+     * @dev Should return whether the signature provided is valid for the provided data
+     * @param hash Hash of the data to be signed
+     * @param signature Signature byte array associated with _data
+     */
+    function isValidSignature(bytes32 hash, bytes memory signature)
+        external
+        view
+        returns (bytes4 magicValue);
+}
+
+/**
+ * @title Pool booster for Merkl distributor
+ * @author Origin Protocol Inc
+ */
+contract PoolBoosterMerkl is IPoolBooster, IERC1271 {
+    /// @notice address of merkl distributor
+    IMerklDistributor public immutable merklDistributor;
+    /// @notice address of the OS token
+    IERC20 public immutable rewardToken;
+    /// @notice if balance under this amount the bribe action is skipped
+    uint256 public constant MIN_BRIBE_AMOUNT = 1e10;
+    /// @notice Campaign duration in seconds
+    uint32 public immutable duration; // -> should be immutable
+    /// @notice Campaign type
+    uint32 public immutable campaignType;
+    /// @notice Merkl hash to sign (for signature verification)
+    bytes32 public immutable hashToSign;
+    /// @notice Owner of the campaign
+    address public immutable creator;
+    /// @notice Campaign data
+    bytes public campaignData;
+
+    constructor(
+        address _rewardToken,
+        address _merklDistributor,
+        uint32 _duration,
+        uint32 _campaignType,
+        bytes32 _hashToSign,
```

Comment:
Instead of passing in the hash from Merkl's `DistributionCreator` contract, lets just call `messageHash` on the `merklDistributor` contract to get it.

This would also mean `setMerklHashToSign` could be replaced with `updateMerklHashToSign`

Reply:
Thinking about this some more, I think we can just drop the `hashToSign` storage variable. Validating the `hash` param in `isValidSignature` doesn't give us much. It's much better to validate that the caller of `isValidSignature` is from the `merklDistributor` contract.

Reply:
I thought about it at the beginning, we could even remove this and fetch it will signing. 
I was worried about Merkl going crazy and make us sign bad stuff. However as this is not really signing, I'm not sure this is a bad thing. 

Should I go for the first option and fetch the message to sign when creating a campaign?

Reply:
> Thinking about this some more, I think we can just drop the `hashToSign` storage variable. Validating the `hash` param in `isValidSignature` doesn't give us much. It's much better to validate that the caller of `isValidSignature` is from the `merklDistributor` contract.

I wrote the previous comment before seeing this answer. I agree that we don't really need to validate the `hash`. I'll remove it.

Reply:
I completely removed `hashToSign` from the contract: [c39d0c1](https://github.com/OriginProtocol/origin-dollar/pull/2474/commits/c39d0c1a224be71f960903008dedd551aea7d6c0)